subprojects/core/src/main/java/org/gradle/plugin/management/ConfigurablePluginRequest.java

```diff
@@ -0,0 +1,17 @@
+package org.gradle.plugin.management;
+
+import org.gradle.api.Incubating;
+
+/**
+ * This needs a better name.
+ *
+ * Allows user to specify details about the plugin that needs
+ * to be loaded.
+ */
+@Incubating
+public interface ConfigurablePluginRequest extends PluginRequest{
+
+    void setVersion(String version);
+
+    void useTarget(String target);
```

Comment:
Should probably take an Object just like dependency resolution rules do. Should be parsed into a `ModuleVersionSelector` immediately, so we don't leak potentially mutable stuff deep into the resolver infrastructure.